crates/oxc_linter/src/rules/typescript/ban_tslint_comment.rs

```diff
@@ -0,0 +1,131 @@
+use lazy_static::lazy_static;
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::{self, Error},
+};
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+use regex::Regex;
+
+use crate::{context::LintContext, fixer::Fix, rule::Rule};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("typescript-eslint(ban-tslint-comment): tslint comment detected: \"{0}\"")]
+#[diagnostic(severity(warning))]
+struct BanTslintCommentDiagnostic(String, #[label] pub Span);
+
+#[derive(Debug, Default, Clone)]
+pub struct BanTslintComment;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule disallows `tslint:<rule-flag>` comments
+    ///
+    /// ### Why is this bad?
+    /// Useful when migrating from TSLint to ESLint. Once TSLint has been
+    /// removed, this rule helps locate TSLint annotations
+    ///
+    /// ### Example
+    /// ```javascript
+    /// // tslint:disable-next-line
+    /// someCode();
+    /// ```
+    BanTslintComment,
+    nursery // since rust regex may not compatible with ECMAScript regex
```

Comment:
I think it's safe to change to `style`

https://typescript-eslint.io/rules/ban-tslint-comment/

> https://typescript-eslint.io/rules/ban-tslint-comment/

Reply:
Changed to style